tests/index.test.ts

```diff
@@ -1,24 +1,116 @@
+const smallFormVariants = '﹨﹩﹪﹫'
+const halfwidthandFullwidthForms = '！Ｄ｢￥￮'
+const alsoHalfwidthandFullwidthForms = 'ｦﾕￇ'
+
+// SMP
+const kanaExtendedB = '𚿵𚿶𚿷𚿸'
+const kanaSupplement = '𛀂𛀀𛀁𛃾𛃿'
+const kanaExtendedA = '𛄀𛄁𛄡𛄢'
+const smallKanaExtension = '𛅕𛄲𛅤𛅥𛅦𛅧'
+const nushu = '𛋸𛋹𛋺𛋻'
+const khitanSmallScript = '𘳒𘳓𘳔𘳕'
+const cjkUnifiedIdeographsExtensionB = '𪛜𪛝𪛞𪛟'
+const cjkUnifiedIdeographsExtensionC = '𫚄𫚅𫜸𫜹'
+const cjkUnifiedIdeographsExtensionD = '𫝀𫝁𫝂𫝃'
+const cjkUnifiedIdeographsExtensionE = '𬉼𬷰𬷱𬺠𬺡'
+const cjkUnifiedIdeographsExtensionF = '𬻿𮨥𮨦𮯠'
+const cjkUnifiedIdeographsExtensionG = '𰀀𰀁𱍀𱍊'
+const cjkUnifiedIdeographsExtensionH = '𱍐𱍑𲎬𲎭'
+const cjkCompatibilityIdeographsSupplement = '乁你鼻𪘀'
+
+const basicLatin = '.a'
+
+function casify(chars: string, t: TestType): TestCases {
+  return Object.fromEntries(chars.split('').map(c => [c, t]))
+}
+
+const moreTestCases: TestCases = {
+  ...casify(cjkRadicalsSupplement, 'cjk-letter'),
+  ...casify(cjkSymbolsPunctuation, 'cjk-punctuation'),
+  ...casify(alsoCjkSymbolsPunctuation, 'cjk-letter'),
+  ...casify(kangxiRadicals, 'cjk-letter'),
+  ...casify(ideographicDescriptionCharacters, 'cjk-punctuation'),
+  ...casify(hiragana + katakana, 'cjk-letter'),
+  ...casify(katakanaPhoneticExtensions, 'cjk-letter'),
+  ...casify(
+    bopomofo + bopomofoExtended + alsoBopomofoExtendedIntroducedInUnicode13,
+    'cjk-letter',
+  ),
+  ...casify(
+    hangulCompatibilityJamo + hangulJamoExtendedA + hangulJamoExtendedB,
+    'cjk-letter',
+  ),
+  ...casify(hangulSyllables, 'cjk-letter'),
+  ...casify(cjkCompatibility, 'cjk-letter'),
+  ...casify(
+    cjkUnifiedIdeographsExtensionA + cjkUnifiedIdeographs,
+    'cjk-letter',
+  ),
+  ...casify(cjkCompatibilityIdeographs, 'cjk-letter'),
+  ...casify(
+    cjkCompatibilityForms + smallFormVariants + halfwidthandFullwidthForms,
+    'cjk-punctuation',
+  ),
+  ...casify(alsoHalfwidthandFullwidthForms, 'cjk-letter'),
+  ...casify(cjkUnifiedIdeographsExtensionE, 'cjk-letter'),
+  ...casify(nushu, 'non-cjk'),
+  ...casify(khitanSmallScript, 'non-cjk'),
+  ...casify(basicLatin, 'non-cjk'),
+}
+
+const shouldBeCjkButCurrentlyNotMatched: TestCases = {
+  ...casify(kanbun, 'non-cjk'),
+  ...casify(cjkStrokes, 'non-cjk'),
+  ...casify(enclosedCjkLettersMonths, 'non-cjk'),
+  ...casify(alsoCjkCompatibility, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionB, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionC, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionD, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionE, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionF, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionG, 'non-cjk'),
+  ...casify(cjkUnifiedIdeographsExtensionH, 'non-cjk'),
+  ...casify(cjkCompatibilityIdeographsSupplement, 'non-cjk'),
+  ...casify(
+    kanaExtendedA + kanaExtendedB + kanaSupplement + smallKanaExtension,
+    'non-cjk',
+  ),
+}
+
+const testCases: TestCases = {
+  ...moreTestCases,
+  ...shouldBeCjkButCurrentlyNotMatched,
```

Comment:
It looks a bit weird that it's named as `moreTestCases`, I guess we could simply place the content of `moreTestCases` and `shouldBeCjkButCurrentlyNotMatched` in `testCases` directly with some comment as separator to indicate "shouldBeCjkButCurrentlyNotMatched"?

Reply:
Since not matched issues has been addressed, there is no need to separate tests as `moreTestCases` and `shouldBeCjkButCurrentlyNotMatched`. Use just one `testCases` instead.